src/Compilers/CSharp/Portable/BoundTree/BoundImplicitIndexerAccess.cs

```diff
@@ -29,5 +29,11 @@ internal BoundExpression GetReceiver()
             Debug.Assert(receiver is not null);
             return receiver;
         }
+
+        private partial void Validate()
+        {
+            Debug.Assert(LengthOrCountAccess is BoundPropertyAccess or BoundLocal);
+            Debug.Assert(IndexerOrSliceAccess is BoundIndexerAccess or BoundCall);
```

Comment:
>IndexerOrSliceAccess is BoundIndexerAccess or BoundCall [](http://example.com/codeflow?start=25&length=55)

This is going to "conflict" with the array PR ([https://github.com/dotnet/roslyn/pull/57918](https://github.com/dotnet/roslyn/pull/57918)). Consider adding BoundArrayAccess into the list.